contents/data-display/charts/weekly-temperature-chart/index.tsx

```diff
@@ -25,48 +30,60 @@ const WeeklyTemperatureChart: FC = () => {
     [(Date | undefined)?, (Date | undefined)?]
   >([today, nextWeek])
 
+  const valueRef = useRef<[(Date | undefined)?, (Date | undefined)?]>([
+    today,
+    nextWeek,
+  ])
+
   const { value, loading } = useAsync(async () => {
-    if (!dateRange[0] || !dateRange[1]) return []
-
-    const startDateString = formatDate(dateRange[0])
-    const endDateString = formatDate(dateRange[1])
-
-    const response = await fetch(
-      `https://api.open-meteo.com/v1/forecast?latitude=35.6785,34.6937,43.0642,26.2124&longitude=139.6823,135.5023,141.3468,127.6792&hourly=temperature_2m&timezone=Asia%2FTokyo&start_date=${startDateString}&end_date=${endDateString}`,
-      { cache: "force-cache" },
-    )
-
-    const data = (await response.json()) as Array<{
-      latitude: number
-      longitude: number
-      hourly: {
-        time: string[]
-        temperature_2m: number[]
-      }
-    }>
-
-    const locationKeys = ["Tokyo", "Osaka", "Hokkaido", "Okinawa"]
-
-    const formattedData = data.reduce(
-      (acc, region, index) => {
-        const locationKey = locationKeys[index]
-        if (!locationKey) {
-          throw new Error("Unexpected location index")
-        }
-        region.hourly.time.forEach((time, i) => {
-          const formattedTime = time.replace("T", " ")
-          if (!acc[i])
-            acc[i] = { date: formattedTime } as Record<string, number | string>
-          acc[i][locationKey] = region.hourly.temperature_2m[i]
-        })
-        return acc
-      },
-      [] as Array<Record<string, number | string>>,
-    )
-
-    return formattedData
+    try {
+      if (!dateRange[0] || !dateRange[1]) return []
+
+      const startDateString = formatDate(dateRange[0])
+      const endDateString = formatDate(dateRange[1])
+
+      return getWeeklyTemperatureData(startDateString, endDateString)
```

Comment:
Don't you use await?

```suggestion
      return await getWeeklyTemperatureData(startDateString, endDateString)
```

Reply:
Fixed.